recipes/atk/all/conanfile.py

```diff
@@ -0,0 +1,88 @@
+from conans import ConanFile, Meson, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import shutil
+
+required_conan_version = ">=1.29"
+
+class LibnameConan(ConanFile):
+    name = "atk"
+    description = "set of accessibility interfaces that are implemented by other toolkits and applications"
+    topics = ("conan", "atk", "accessibility")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.atk.org"
+    license = "LGPL-2.1-or-later"
+    generators = "pkg_config"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        }
+    default_options = {
+        "shared": True,
```

Comment:
any restriction that only requires shared by default?

Reply:
It's only because the upstream project is built as shared library by default, [as all meson projects](https://mesonbuild.com/Builtin-options.html#core-options).

Reply:
The problem with mixing static and shared libraries is that the static dependencies of this library are built-in.
So if another library also uses a static libraries, then the glib symbols are duplicated.

Reply:
yes, but this problem existed long before conan, and we certainly take less risk by keeping the same configuration as what everybody has been doing. [glib is also shared by default](https://github.com/conan-io/conan-center-index/blob/master/recipes/glib/all/conanfile.py#L22)

Reply:
glib is wrong, I think.
it requires libffi and zlib, which are static by default.
So libffi and zlib are builtin.

glib allows static libraries, as does atk.
And cci has apparantly chosen to have static libraries by default.
So I think these (=glib and atk) should be static too by default.

I think your reason *It's because the upstream project is built as shared library by default* is moot because almost all autotools project build shared libraries by default.

Reply:
that's is not first project using meson. I prefer keeping is as static, because is not a real restriction, is only a preference due meson. CCI uses static by, therefore, keep static.

Reply:
I already changed it in https://github.com/conan-io/conan-center-index/pull/2301/commits/fb28c6affe8e960420e24017437b6646a72b4872#diff-488fff635176b5fd8844254b23e02587L23 ;)